test/test_dos.py

```diff
@@ -5434,6 +5434,386 @@ def test_fat_ds3_share_open_twice(self):
+  enum {
+    DELFCB,
+    RENFCB,
+    DELPTH,
+    RENPTH,
+  } testmode;
+
+  if (argc < 6) {
+    printf("FAIL: Missing arguments (primary|secondary) prismode priomode expected (DELFCB|RENFCB|DELPTH|RENPTH)\n");
+    return -2;
+  }
+
+  if (strcmp(argv[1], "primary") == 0)
+    primary = 1;
+  if (strcmp(argv[1], "secondary") == 0)
+    primary = 0;
+  if (primary < 0) {
+    printf("FAIL: Invalid argument (primary|secondary)\n");
+    return -2;
+  }
+
+  prismode = sharemode(argv[2]);
+  if (prismode == 0xff) {
+    printf("FAIL: Invalid argument prismode '%s'\n", argv[2]);
+    return -2;
+  }
+  priomode = openmode(argv[3]);
+  if (priomode == 0xff) {
+    printf("FAIL: Invalid argument priomode '%s'\n", argv[3]);
+    return -2;
+  }
+
+  // expected result is argv[4]
+
+  if (strcmp(argv[5], "DELFCB") == 0)
+    testmode = DELFCB;
+  else if (strcmp(argv[5], "RENFCB") == 0)
+    testmode = RENFCB;
+  else if (strcmp(argv[5], "DELPTH") == 0)
+    testmode = DELPTH;
+  else if (strcmp(argv[5], "RENPTH") == 0)
+    testmode = RENPTH;
+  else {
+    printf("FAIL: Invalid argument testmode '%s'\n", argv[5]);
+    return -2;
+  }
+
+  // Print results in consistent format
+  // FAIL:("SH_DENYNO", "RW", "ALLOW")[secondary denied]
+
+  if (primary) {
+    unsigned short mode = prismode | priomode;
+
+    // create the file
+    ret = _dos_creat(FN1 "." FE1, _A_NORMAL, &handle);
+    if (ret != 0) {
+      printf("FAIL:('%s', '%s', '%s')[primary create]\n", argv[2], argv[3], argv[4]);
+      return -1;
+    }
+    _dos_close(handle);
+
+    // remove the target file for rename
+    unlink(FN2 "." FE2);
+
+    // open with correct flags
+    ret = _dos_open(FN1 "." FE1, mode, &handle);
+    if (ret != 0) {
+      printf("FAIL:('%s', '%s', '%s')[primary denied]\n", argv[2], argv[3], argv[4]);
+      return -1;
+    }
+//    printf("INFO: primary: File was opened with mode 0x%04x\n", mode);
+
+    // Now start second copy
+    spawnlp(P_WAIT, argv[0], argv[0], "secondary", argv[2], argv[3], argv[4], argv[5], NULL);
+
+    _dos_close(handle);
+
+  } else { // secondary
+    switch(testmode) {
+
+      case DELFCB: {
+        /*
+          DOS 1+ - DELETE FILE USING FCB
+
+          AH = 13h
+          DS:DX -> unopened FCB (see #01345), filename filled with template for
+          deletion ('?' wildcards allowed)
+
+          Return:
+            AL = status
+              00h one or more files successfully deleted
+              FFh no matching files or all were read-only or locked
+        */
+
+        struct {
+          uint8_t drive;
+          char name[8];
+          char ext[3];
+          char padding[25];
+        } fcb;
+        uint16_t ax;
+
+        fcb.drive = 0;
+        memcpy(&fcb.name, FN1, 8);
+        memcpy(&fcb.ext, FE1, 3);
+
+        asm volatile("int $0x21\n"
+                     : "=a"(ax)
+                     : "a"(0x1305), "d"(&fcb)
+                     : /* nothing clobbered */ );
```

Comment:
Note that if you pass the pointer
as input (&fcb) and it can be modified,
you need to clobber "memory".

Reply:
Is that also true if it's modified, but I don't care about its new contents?

Reply:
It depends.
If you don't use it at all after
that, maybe its not important.
But otherwise it will be treated
as an uninitialized space.

Reply:
My DOS book doesn't mention that int21/13 is modified, so am I right in saying I don't need to add the `memory` keyword here?

Reply:
Yes, for read-only its fine of course.

Reply:
Thanks.